src/script/lua_api/l_object.cpp

```diff
@@ -1744,26 +1869,250 @@ int ObjectRef::l_get_sky(lua_State *L)
 	NO_MAP_LOCK_REQUIRED;
 	ObjectRef *ref = checkobject(L, 1);
 	RemotePlayer *player = getplayer(ref);
-	if (player == NULL)
+
+	if (!player)
 		return 0;
-	video::SColor bgcolor(255, 255, 255, 255);
-	std::string type;
-	std::vector<std::string> params;
-	bool clouds;
+	
+	const SkyboxParams &skybox_params = player->getSkyParams();
+
+	lua_newtable(L);
+	push_ARGB8(L, skybox_params.bgcolor);
+	lua_setfield(L, -2, "base_color");
+	lua_pushstring(L, skybox_params.type.c_str());
+	lua_setfield(L, -2, "type");
+
+	// Add textures to a numerically indexed table for clean dump() prints
+	// We also don't want to show it when we're not using the textured skybox
+	if (skybox_params.params[0] == "" || skybox_params.type != "skybox") {
+		lua_pushnil(L);
+	} else {
+		lua_newtable(L);
+		for (int i = 0; i < 6; i++) {
+			// Create table key
+			lua_pushnumber(L, i+1);
+			lua_pushstring(L, skybox_params.params[i].c_str());
+			lua_settable(L, -3);
+		}
+		lua_setfield(L, -2, "textures");
+	}
 
-	player->getSky(&bgcolor, &type, &params, &clouds);
-	type = type.empty() ? "regular" : type;
+	lua_pushboolean(L, skybox_params.clouds);
+	lua_setfield(L, -2, "clouds");
 
-	push_ARGB8(L, bgcolor);
-	lua_pushlstring(L, type.c_str(), type.size());
+	// Create table containing modified default skybox color
+	// and sun, moon sun[rise/set] tints.
 	lua_newtable(L);
-	s16 i = 1;
-	for (const std::string &param : params) {
-		lua_pushlstring(L, param.c_str(), param.size());
-		lua_rawseti(L, -2, i++);
+	if (skybox_params.tint_type == "regular") {
+		push_ARGB8(L, skybox_params.day_sky);
+		lua_setfield(L, -2, "day_sky");
+		push_ARGB8(L, skybox_params.day_horizon);
+		lua_setfield(L, -2, "day_horizon");
+		push_ARGB8(L, skybox_params.dawn_sky);
+		lua_setfield(L, -2, "dawn_sky");
+		push_ARGB8(L, skybox_params.dawn_horizon);
+		lua_setfield(L, -2, "dawn_horizon");
+		push_ARGB8(L, skybox_params.night_sky);
+		lua_setfield(L, -2, "night_sky");
+		push_ARGB8(L, skybox_params.night_horizon);
+		lua_setfield(L, -2, "night_horizon");
+		push_ARGB8(L, skybox_params.indoors);
+		lua_setfield(L, -2, "indoors");
 	}
-	lua_pushboolean(L, clouds);
-	return 4;
+	
+	push_ARGB8(L, skybox_params.sun_tint);
+	lua_setfield(L, -2, "sun_tint");
+	push_ARGB8(L, skybox_params.moon_tint);
+	lua_setfield(L, -2, "moon_tint");
+	lua_pushstring(L, skybox_params.tint_type.c_str());
+	lua_setfield(L, -2, "tint_type");
+	lua_setfield(L, -2, "sky_color");
```

Comment:
missing value?

Reply:
https://gist.github.com/Jordach/f34d75ee2a75226087e950cdfc76ac49

Dumped output of get_sun/moon/stars/sky.

Reply:
oh, it's a table